packages/components/src/date-time/time.js

```diff
@@ -189,52 +191,62 @@ class TimePicker extends Component {
 		this.setState( { minutes: event.target.value } );
 	}
 
+	renderMonth( month ) {
+		return (
+			<div className="components-datetime__time-field components-datetime__time-field-month">
+				<select
+					aria-label={ __( 'Month' ) }
+					className="components-datetime__time-field-month-select"
+					value={ month }
+					onChange={ this.onChangeMonth }
+					onBlur={ this.updateMonth }
+				>
+					<option value="01">{ __( 'January' ) }</option>
+					<option value="02">{ __( 'February' ) }</option>
+					<option value="03">{ __( 'March' ) }</option>
+					<option value="04">{ __( 'April' ) }</option>
+					<option value="05">{ __( 'May' ) }</option>
+					<option value="06">{ __( 'June' ) }</option>
+					<option value="07">{ __( 'July' ) }</option>
+					<option value="08">{ __( 'August' ) }</option>
+					<option value="09">{ __( 'September' ) }</option>
+					<option value="10">{ __( 'October' ) }</option>
+					<option value="11">{ __( 'November' ) }</option>
+					<option value="12">{ __( 'December' ) }</option>
+				</select>
+			</div>
+		);
+	}
+
+	renderDay( day ) {
+		return (
+			<div className="components-datetime__time-field components-datetime__time-field-day">
+				<input
+					aria-label={ __( 'Day' ) }
+					className="components-datetime__time-field-day-input"
+					type="number"
+					value={ day }
+					step={ 1 }
+					min={ 1 }
+					onChange={ this.onChangeDay }
+					onBlur={ this.updateDay }
+				/>
+			</div>
+		);
+	}
+
 	render() {
 		const { is12Hour } = this.props;
 		const { day, month, year, minutes, hours, am } = this.state;
+		const renderMonthDay = [ this.renderMonth( month ), this.renderDay( day ) ];
+		const renderDayMonth = [ this.renderDay( day ), this.renderMonth( month ) ];
```

Comment:
I hadn't considered it initializing those elements when the variable is assigned. I'll have to find a way to make it so that only one is instantiated when necessary.

Reply:
Yeah, I like the idea of using `reverse`.

The warning is In the browser console—you'll need SCRIPT_DEBUG enabled in your dev environment to see it:
https://codex.wordpress.org/Debugging_in_WordPress

Some details about the particular warning from React's docs:
https://reactjs.org/docs/lists-and-keys.html

Reply:
Thanks @talldan I'll look into this and see what is going on! I appreciate all the help.

Reply:
I've made a few changes and I believe I also fixed the issue with the array keys. I looked at a few other components and created a simple key for each root element that is returned for both the day and month render methods.